config/e2e/monitoring.yaml

```diff
@@ -1,408 +1,410 @@
-    - add_fields:
-        target: ''
-        fields:
-          pipeline: {{ .Pipeline }}
-          build_number: {{ .BuildNumber }}
-          provider: {{ .Provider }}
-          clusterName: {{ .ClusterName }}
-          kubernetes_version: {{ .KubernetesMajorMinor }}
-          stack_version: {{ .ElasticStackVersion }}
-          e2e_test_id: {{ .Pipeline }}-{{ .BuildNumber }}-{{ .Provider }}-{{ .ClusterName }}-{{ .KubernetesMajorMinor }}-{{ .ElasticStackVersion }}
-    setup.template.overwrite: true
-    setup.template.append_fields:
-    - name: stack_version
-      type: keyword
-    - name: kubernetes_version
-      type: keyword
-    - name: build_number
-      type: keyword
-    - name: pipeline
-      type: keyword
-    - name: e2e_test_id
-      type: keyword
-    - name: provider
-      type: keyword
-    - name: clusterName
-      type: keyword
-    output.elasticsearch:
-      hosts: ['${monitoring_url}']
-      username: ${monitoring_user}
-      password: ${monitoring_pass}
-      # Since E2E cluster's Elasticsearch instance may lag behind the newest version of Beat, we need to allow this scenario.
-      allow_older_versions: true
-      # The maximum number of events to bulk in a single Elasticsearch bulk API index request. Needed if Elasticsearch is behind Cloud Ingress.
-      bulk_max_size: 10
+  version: 7.15.1 # needs to less or equal to e2e-monitor version
+  elasticsearchRefs:
+    - secretName: eck-{{ .TestRun }}
   daemonSet:
     podTemplate:
       spec:
-        serviceAccountName: metricbeat
-        automountServiceAccountToken: true # some older Beat versions are depending on this settings presence in k8s context
-        initContainers:
-        - name: elastic-internal-init-keystore
-          securityContext:
-           {{ if .OcpCluster }}
-           privileged: true
-           {{ end }}
-           runAsUser: 0
+        automountServiceAccountToken: true
+        serviceAccountName: e2e-agent
         containers:
-        - args:
-          - -e
-          - -c
-          - /etc/beat.yml
-          - -system.hostfs=/hostfs
-          name: metricbeat
-          volumeMounts:
-          - mountPath: /hostfs/sys/fs/cgroup
-            name: cgroup
-          - mountPath: /var/run/docker.sock
-            name: dockersock
-          - mountPath: /hostfs/proc
-            name: proc
-          env:
-          - name: NODE_NAME
-            valueFrom:
-              fieldRef:
-                fieldPath: spec.nodeName
-        dnsPolicy: ClusterFirstWithHostNet
-        hostNetwork: true # Allows to provide richer host metadata
-        securityContext:
-          runAsUser: 0
-        terminationGracePeriodSeconds: 30
+          - name: agent
+            securityContext:
+              runAsUser: 0
+            resources:
+              limits:
+                memory: 768Mi
+                cpu: 1
+              requests:
+                memory: 768Mi
+                cpu: 100m
```

Comment:
It's a good thing to take the opportunity to increase memory because Filebeat was regularly OOMkilled.